### ISSUE TITLE
[discussion] Branch management

### ISSUE THREAD

**Issue by TDW89** [KSP-KOS/KSLib#72]
Discussion to work out what branches we need, what to call them and how we are going to organise them.
@space-is-hard - tagging you to make sure you see this.

**Comment by space-is-hard** [comment]
This is going to have to be where our back-and-forth takes place today; apologies in advance.

I think first we need to outline what kind of structure we want KSLib to take. Do we want:

1. A "package" that gets updated in discrete increments; i.e. push out a release every now and then that's got new stuff that's been tested. Users would be encouraged to download the whole package and use what they need. This would require a develop branch to work with and a master branch to release from.

2. A continuously updating collection; i.e. library additions or improvements are constantly pushed to master after testing them. This would arguably only require a master branch.

3. Something else? Open to ideas.

**Comment by TDW89** [comment]
Not really sure on this The continuous updates would probably be easier but people using this library (other than those contributing) probably wont check back very often (if at all) for updates if they are lots of small things.
If we want a way of informing people of updates then they should probably be larger updates.
I don't use ckan but we could potentially stick the kslib in as a recommended/ suggested download in the same way that the extra contract packs are for contract configerator.

**Comment by space-is-hard** [comment]
Or the latest version of KSLib could be included in the kOS download, which I think would serve a dual-purpose of giving the user some useful functions for stuff *and* letting new users see examples of the code and of functions in general

**Comment by TDW89** [comment]
In which case discreet updates are probably best.
We should also probably make a point of creating some scripts that use the newest language features for a release so to increase visibility of those features (for people that might not read the change log)

**Comment by space-is-hard** [comment]
Agreed. Master already seems to be up-to-date, let's go ahead and get some of our PR backlog pushed to develop, test it, and push it to master for release of KSLib v0.1(?)

**Comment by TDW89** [comment]
Ok i'm going to start going through the testing the PR's I will leave a comment on them once they are tested. If you could have a look at them for spelling etc and leave a comment if you get to that one first, or merge if i have already tested. (who ever gets there second merges it)

**Comment by space-is-hard** [comment]
Sounds good

**Comment by TDW89** [comment]
before we release we should probably go through an make sure everything has the 
```
// This file is distributed under the terms of the MIT license, (c) the KSLib team
```
line in the header too.

**Comment by space-is-hard** [comment]
The develop branch is way behind master; I don't think we need the develop branch anyways since KSLib is in a state of "continuous release". Does anybody have objections to deleting the develop branch?
